src/hdmint/wallet.cpp

```diff
@@ -249,19 +251,21 @@ void CHDMintWallet::SetWalletTransactionBlock(CWalletTx &wtx, const CBlockIndex
 void CHDMintWallet::SyncWithChain(bool fGenerateMintPool, boost::optional<std::list<std::pair<uint256, MintPoolEntry>>> listMints)
 {
     CWalletDB walletdb(strWalletFile);
-    bool foundSigma = true;
-    bool foundLela = true;
-
+    bool found = true;
 
     std::set<uint256> setAddedTx;
     std::set<uint256> setChecked;
-    while (foundSigma || foundLela) {
-        foundSigma = false;
-        foundLela = false;
+    int mintsFound = 1;
+    bool firstIteration = true;
+    while (found || mintsFound > 0) {
+        found = false;
+        mintsFound = 0;
         if (fGenerateMintPool)
-            GenerateMintPool(walletdb);
+            GenerateMintPool(walletdb, !firstIteration);
```

Comment:
Please clarify why the first iteration is different

Reply:
1. we need to generate new mintpool when we find out that we have used entry in current one, 
2. in each startup we are running this function to update our mint state, 
3 we are preventing generation at first iteration as we dont know the answer of 1, so we are running sync process for current mintpool and then if we find used mints, we are generating a new one